binding/python/docs/TUTORIAL.md

```diff
@@ -0,0 +1,144 @@
+
+Here is a typical example of python code with multvierso.
+```python
+# import multiverso.
+import multiverso as mv
+# Init multiverso.
+mv.init()
+# Get total number of workers.
+print mv.workers_num()
+# Get the id for current worker.
+print mv.worker_id()
+# if the worker is master worker.
+print mv.is_master_worker()
+
+# Here is your code to create tables, train models and sync values.
+
+# Shutdown multiverso
+mv.shutdown()
+```
+
+## About the master worker
+Some things only need one worker process, such as validation, outputting the result, initializing the parameters and so on. So we mark the worker 0 as the master worker to finish these things. You can use `mv.is_master_worker` to distinguish the master worker from others.
+For example, if you want to make sure only one process will initialize the parameters, you can write similar code below.
+```
+import multiverso as mv
+# create your table handler tbh and initial value params
+if mv.is_master_worker():
+    # Only master worker will set the initial value.
+    tbh.add(params)
+    # Set a barrier for other workers to wait.
+    mv.barrier()
+else:
+    # Wait the master worker to finish setting.
+    mv.barrier()
+    # Get the initial model from the server.
+    params = tbh.get()
+```
+
+There are similar strategies in `theano_ext.sharedvar` and `lasagne_ext.param_manager` when initializing values. They are already implemented in the constructors.
+
+
+
+# How to use multiverso in theano
+First, make sure you have understood `mv.init()`, `mv.shutdown()` and `mv.barrier()` mentioned in last section.  You should add the same functions in your theano python script.
+
+In theano, parameters are usually stored in sharedVariables.
+
+For example, sharedVariables can be created like this in a theano script.
+```
+self.W = theano.shared(
+    value=numpy.zeros(
+        (n_in, n_out),
+        dtype=theano.config.floatX
+    ),
+    name='W',
+    borrow=True
+)
+```
+
+If you want to use multiverso, you can modify them like this.
+```
+from multiverso.theano_ext import sharedvar
+# Only the master worker can initialize the shared value. The initial value from
+# other processes will be ignored
+W = sharedvar.mv_shared(
+    value=numpy.zeros(
+        (n_in, n_out),
+        dtype=theano.config.floatX
+    ),
+    name='W',
+    borrow=True
+)
+
+# build the model
+
+# train the modle
+
+# When you are ready to add the delta of the variable to parameter server and sync the latest value, you can run this function
+W.mv_sync()
+
+
+# If you want to sync all variables created by `sharedvar.mv_shared`, you can use this function
+sharedvar.sync_all_mv_shared_vars()
+```
+
+`mv_shared` is just a wrapper of `theano.shared`. It acts same as `theano.shared`, but it give you much more convenient interface to sync values.
```

Comment:
It acts the same as `theano.shared` while making it more convenient to sync values.